src/iosrv/obj.c

```diff
@@ -38,24 +36,139 @@
 #include <daos_task.h>
 
 #include <daos_srv/daos_server.h>
+#include "srv_internal.h"
+
+/*
+ * TODO:
+ * Client APIs may need to acquire some global pthread lock, that could block
+ * the whole xstream unexpectedly, we need to revise the client APIs to make
+ * sure the global phtread locks are not used when they are called on server.
+ */
+static void
+dsc_progress(void *arg)
+{
+	struct dss_xstream	*dx = arg;
+
+	while (!dss_in_shutdown(dx)) {
+		tse_sched_progress(&dx->dx_sched_dsc);
+		ABT_thread_yield();
+	}
+}
+
+static int
+dsc_progress_start(void)
+{
+	struct dss_xstream	*dx = dss_get_module_info()->dmi_xstream;
+	int			 rc;
+
+	if (dx->dx_dsc_started)
+		return 0;
+
+	rc = ABT_thread_create(dx->dx_pools[DSS_POOL_REBUILD], dsc_progress,
```

Comment:
It may be more general if input the @type via the dsc_progress_start() parameter instead of always setting it as "DSS_POOL_REBUILD", that may be helpful for other use cases in the future.

Reply:
There is only one tse scheduler and one dsc ULT per xstream, so I don't think passing a @type is meaningful.